### ISSUE TITLE
Comparison of gpu acceleration between gtx 1660ti(6g video memory) and quadro gv100(32g video memory)

### ISSUE THREAD

**Issue by lihualilee** [tensorflow/tensorflow#44000]
I use the same mnist data set code to classify on gtx 1660ti graphics card and Quadro gv100 graphics card, and test that GPU acceleration is available.
Why is the speed on Quadro gv100 slower than gtx 1660ti, and what conditions should be met to turn on gpu acceleration on Quadro gv100? But I have verified that gpu acceleration is available.
![image](https://user-images.githubusercontent.com/64268860/95950710-a635f700-0e27-11eb-9116-5585afaf5f65.png)
Fig. 1 is the experimental result (37s) on gtx 1660ti
![image](https://user-images.githubusercontent.com/64268860/95955723-ded9ce80-0e2f-11eb-9bd5-2f145a1214c2.png)
![image](https://user-images.githubusercontent.com/64268860/95955752-ec8f5400-0e2f-11eb-8073-0f92795295c6.png)
Fig. 2 and 3 is the result of the experiment on Quadro gv100
![image](https://user-images.githubusercontent.com/64268860/95955860-12b4f400-0e30-11eb-84fe-0972181df3a9.png)

Fig. 4 shows the memory usage of Quadro gv100
Because of this problem, I trained very slowly when I used a lot of data to do semantic segmentation experiments. I don't think this Quadro gv100 graphics card should be trained so slowly.
Thank you very much for helping me!

**Comment by ravikyram** [comment]
@lihualilee 

Please, help us with code snippet instead of screenshot to reproduce the issue in our environment.It helps us in localizing the issue faster.Thanks!

**Comment by lihualilee** [comment]
We conducted experiments on two different computers. One computer's GPU is Quadro GV100 and the other is gtx 1660ti. We run the same code under the same computer. The code is as follows:`import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers
print(tf.test.is_gpu_available())
    """
    WARNING:tensorflow:From <ipython-input-1-5819ce0d6808>:4: is_gpu_available (from tensorflow.python.framework.test_util) is deprecated and will be removed in a future version.
    Instructions for updating:
    Use `tf.config.list_physical_devices('GPU')` instead.
    True
    """
(x_train, y_train), (x_test, y_test) = keras.datasets.mnist.load_data()

x_train = x_train.reshape(60000,28,28,1).astype("float32") / 255
x_test = x_test.reshape(10000,28,28,1).astype("float32") / 255

y_train = y_train.astype("float32")
y_test = y_test.astype("float32")
inputs = keras.Input(shape=(28,28,1), name="digits")

x= layers.Conv2D(32,kernel_size=3,strides=1,padding="same",activation='relu')(inputs)
x= layers.Conv2D(64,kernel_size=3,strides=1,padding="same",activation='relu')(x)

x = layers.MaxPooling2D(pool_size=2,strides=2,padding='same')(x)

x = layers.Conv2D(128,kernel_size=3,strides=1,padding="same",activation='relu')(x)
x = layers.Conv2D(256,kernel_size=3,strides=1,padding="same",activation='relu')(x)
x = layers.MaxPooling2D(pool_size=2,strides=2,padding='same')(x)

x = layers.Conv2D(512,kernel_size=5,strides=1,padding="same",activation='relu')(x)
x = layers.Conv2D(1024,kernel_size=5,strides=1,padding="same",activation='relu')(x)
x = layers.GlobalAveragePooling2D()(x)

x = layers.Dense(64, activation="relu", name="dense_1")(x)
x = layers.Dense(64, activation="relu", name="dense_2")(x)
output= layers.Dense(10)(x)
outputs = layers.Activation('softmax', dtype='float32')(output)
model = keras.Model(inputs=inputs, outputs=outputs)
# model.summary()
model.compile( optimizer="adam", loss=keras.losses.SparseCategoricalCrossentropy(),metrics=['acc'])

history = model.fit(x_train,y_train,batch_size=256,epochs=20,validation_data=(x_test, y_test))
      “”“
       Epoch 1/20
       235/235 [==============================] - 139s 593ms/step - loss: 0.6885 - acc: 0.7536 - 
      val_loss: 0.0754 - val_acc: 0.9765
      Epoch 2/20
       235/235 [==============================] - 133s 568ms/step - loss: 0.0652 - acc: 0.9805 - 
       val_loss: 0.0290 - val_acc: 0.9902
      Epoch 3/20
”“”

`

I use the same mnist data set code to classify on gtx 1660ti graphics card and quadro gv100 graphics card, and test that GPU acceleration is available.
Why is the speed on quadro gv100 slower than gtx 1660ti? It takes 240 seconds to train an epoch when conducting experiments on Quadro gv100.
It takes 37 seconds to train an epoch in the experiment on gtx 1660. The performance of Quqdro gv100 should be better than that of gtx 1660ti.
Why did I get a completely different point of view when I went to the lab? quadro gv100 is so much slower than gtx 1660ti.
What conditions should be met to turn on gpu acceleration on quadro gv100? But I have verified that gpu acceleration is available

**Comment by lihualilee** [comment]
We conducted experiments on two different computers. One computer's GPU is Quadro GV100 and the other is gtx 1660ti. We run the same code under the same computer. The code is as follows:
import tensorflow as tf 
from tensorflow import keras
 from tensorflow.keras import layers 
print(tf.test.is_gpu_available()) 
      """ WARNING:tensorflow:From <ipython-input-1-5819ce0d6808>:4: is_gpu_available (from 
       tensorflow.python.framework.test_util) is deprecated and will be removed in a future version. Instructions for 
        updating: Use tf.config.list_physical_devices('GPU')` instead.
        True
        """
(x_train, y_train), (x_test, y_test) = keras.datasets.mnist.load_data()

x_train = x_train.reshape(60000,28,28,1).astype("float32") / 255
x_test = x_test.reshape(10000,28,28,1).astype("float32") / 255

y_train = y_train.astype("float32")
y_test = y_test.astype("float32")
inputs = keras.Input(shape=(28,28,1), name="digits")

x= layers.Conv2D(32,kernel_size=3,strides=1,padding="same",activation='relu')(inputs)
x= layers.Conv2D(64,kernel_size=3,strides=1,padding="same",activation='relu')(x)

x = layers.MaxPooling2D(pool_size=2,strides=2,padding='same')(x)

x = layers.Conv2D(128,kernel_size=3,strides=1,padding="same",activation='relu')(x)
x = layers.Conv2D(256,kernel_size=3,strides=1,padding="same",activation='relu')(x)
x = layers.MaxPooling2D(pool_size=2,strides=2,padding='same')(x)

x = layers.Conv2D(512,kernel_size=5,strides=1,padding="same",activation='relu')(x)
x = layers.Conv2D(1024,kernel_size=5,strides=1,padding="same",activation='relu')(x)
x = layers.GlobalAveragePooling2D()(x)

x = layers.Dense(64, activation="relu", name="dense_1")(x)
x = layers.Dense(64, activation="relu", name="dense_2")(x)
output= layers.Dense(10)(x)
outputs = layers.Activation('softmax', dtype='float32')(output)
model = keras.Model(inputs=inputs, outputs=outputs)

model.summary()
model.compile( optimizer="adam", loss=keras.losses.SparseCategoricalCrossentropy(),metrics=['acc'])

history = model.fit(x_train,y_train,batch_size=256,epochs=20,validation_data=(x_test, y_test))
“”“
Epoch 1/20
235/235 [==============================] - 139s 593ms/step - loss: 0.6885 - acc: 0.7536 -
val_loss: 0.0754 - val_acc: 0.9765
Epoch 2/20
235/235 [==============================] - 133s 568ms/step - loss: 0.0652 - acc: 0.9805 -
val_loss: 0.0290 - val_acc: 0.9902
Epoch 3/20
”“”

`

I use the same mnist data set code to classify on gtx 1660ti graphics card and quadro gv100 graphics card, and test that GPU acceleration is available.
Why is the speed on quadro gv100 slower than gtx 1660ti? It takes 240 seconds to train an epoch when conducting experiments on Quadro gv100.
It takes 37 seconds to train an epoch in the experiment on gtx 1660. The performance of Quqdro gv100 should be better than that of gtx 1660ti.
Why did I get a completely different point of view when I went to the lab? quadro gv100 is so much slower than gtx 1660ti.
What conditions should be met to turn on gpu acceleration on quadro gv100? But I have verified that gpu acceleration is available

**Comment by lihualilee** [comment]
The operating system of Quadro GV100 is windows server 2016, and the operating system of gtx 1660ti is windows10

**Comment by lihualilee** [comment]
> @lihualilee
> 
> Please, help us with code snippet instead of screenshot to reproduce the issue in our environment.It helps us in localizing the issue faster.Thanks!
We conducted experiments on two different computers. One computer's GPU is Quadro GV100 and the other is gtx 1660ti. 
The operating system of Quadro GV100 is windows server 2016, and the operating system of gtx 1660ti is windows10
We run the same code under the same computer. The code is as follows:
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers
print(tf.test.is_gpu_available())
""" WARNING:tensorflow:From :4: is_gpu_available (from
tensorflow.python.framework.test_util) is deprecated and will be removed in a future version. Instructions for
updating: Use tf.config.list_physical_devices('GPU')` instead.
True
"""
(x_train, y_train), (x_test, y_test) = keras.datasets.mnist.load_data()

x_train = x_train.reshape(60000,28,28,1).astype("float32") / 255
x_test = x_test.reshape(10000,28,28,1).astype("float32") / 255

y_train = y_train.astype("float32")
y_test = y_test.astype("float32")
inputs = keras.Input(shape=(28,28,1), name="digits")

x= layers.Conv2D(32,kernel_size=3,strides=1,padding="same",activation='relu')(inputs)
x= layers.Conv2D(64,kernel_size=3,strides=1,padding="same",activation='relu')(x)

x = layers.MaxPooling2D(pool_size=2,strides=2,padding='same')(x)

x = layers.Conv2D(128,kernel_size=3,strides=1,padding="same",activation='relu')(x)
x = layers.Conv2D(256,kernel_size=3,strides=1,padding="same",activation='relu')(x)
x = layers.MaxPooling2D(pool_size=2,strides=2,padding='same')(x)

x = layers.Conv2D(512,kernel_size=5,strides=1,padding="same",activation='relu')(x)
x = layers.Conv2D(1024,kernel_size=5,strides=1,padding="same",activation='relu')(x)
x = layers.GlobalAveragePooling2D()(x)

x = layers.Dense(64, activation="relu", name="dense_1")(x)
x = layers.Dense(64, activation="relu", name="dense_2")(x)
output= layers.Dense(10)(x)
outputs = layers.Activation('softmax', dtype='float32')(output)
model = keras.Model(inputs=inputs, outputs=outputs)

model.summary()
model.compile( optimizer="adam", loss=keras.losses.SparseCategoricalCrossentropy(),metrics=['acc'])

history = model.fit(x_train,y_train,batch_size=256,epochs=20,validation_data=(x_test, y_test))
“”“
Epoch 1/20
235/235 [==============================] - 139s 593ms/step - loss: 0.6885 - acc: 0.7536 -
val_loss: 0.0754 - val_acc: 0.9765
Epoch 2/20
235/235 [==============================] - 133s 568ms/step - loss: 0.0652 - acc: 0.9805 -
val_loss: 0.0290 - val_acc: 0.9902
Epoch 3/20
”“”

`

I use the same mnist data set code to classify on gtx 1660ti graphics card and quadro gv100 graphics card, and test that GPU acceleration is available.
Why is the speed on quadro gv100 slower than gtx 1660ti? It takes 240 seconds to train an epoch when conducting experiments on Quadro gv100.
It takes 37 seconds to train an epoch in the experiment on gtx 1660. The performance of Quqdro gv100 should be better than that of gtx 1660ti.
Why did I get a completely different point of view when I went to the lab? quadro gv100 is so much slower than gtx 1660ti.
What conditions should be met to turn on gpu acceleration on quadro gv100? But I have verified that gpu acceleration is available

The operating system of Quadro GV100 is windows server 2016, and the operating system of gtx 1660ti is windows10

**Comment by lihualilee** [comment]
@ravikyram 
Please, help us with code snippet instead of screenshot to reproduce the issue in our environment.It helps us in localizing the issue faster.Thanks!
We conducted experiments on two different computers. One computer's GPU is Quadro GV100 and the other is gtx 1660ti.
The operating system of Quadro GV100 is windows server 2016, and the operating system of gtx 1660ti is windows10
We run the same code under the same computer. The code is as follows:
import tensorflow as tf
from tensorflow import keras
from tensorflow.keras import layers
print(tf.test.is_gpu_available())
""" WARNING:tensorflow:From :4: is_gpu_available (from
tensorflow.python.framework.test_util) is deprecated and will be removed in a future version. Instructions for
updating: Use tf.config.list_physical_devices('GPU')` instead.
True
"""
(x_train, y_train), (x_test, y_test) = keras.datasets.mnist.load_data()

x_train = x_train.reshape(60000,28,28,1).astype("float32") / 255
x_test = x_test.reshape(10000,28,28,1).astype("float32") / 255

y_train = y_train.astype("float32")
y_test = y_test.astype("float32")
inputs = keras.Input(shape=(28,28,1), name="digits")

x= layers.Conv2D(32,kernel_size=3,strides=1,padding="same",activation='relu')(inputs)
x= layers.Conv2D(64,kernel_size=3,strides=1,padding="same",activation='relu')(x)

x = layers.MaxPooling2D(pool_size=2,strides=2,padding='same')(x)

x = layers.Conv2D(128,kernel_size=3,strides=1,padding="same",activation='relu')(x)
x = layers.Conv2D(256,kernel_size=3,strides=1,padding="same",activation='relu')(x)
x = layers.MaxPooling2D(pool_size=2,strides=2,padding='same')(x)

x = layers.Conv2D(512,kernel_size=5,strides=1,padding="same",activation='relu')(x)
x = layers.Conv2D(1024,kernel_size=5,strides=1,padding="same",activation='relu')(x)
x = layers.GlobalAveragePooling2D()(x)

x = layers.Dense(64, activation="relu", name="dense_1")(x)
x = layers.Dense(64, activation="relu", name="dense_2")(x)
output= layers.Dense(10)(x)
outputs = layers.Activation('softmax', dtype='float32')(output)
model = keras.Model(inputs=inputs, outputs=outputs)

model.summary()
model.compile( optimizer="adam", loss=keras.losses.SparseCategoricalCrossentropy(),metrics=['acc'])

history = model.fit(x_train,y_train,batch_size=256,epochs=20,validation_data=(x_test, y_test))
“”“
Epoch 1/20
235/235 [==============================] - 139s 593ms/step - loss: 0.6885 - acc: 0.7536 -
val_loss: 0.0754 - val_acc: 0.9765
Epoch 2/20
235/235 [==============================] - 133s 568ms/step - loss: 0.0652 - acc: 0.9805 -
val_loss: 0.0290 - val_acc: 0.9902
Epoch 3/20
”“”

I use the same mnist data set code to classify on gtx 1660ti graphics card and quadro gv100 graphics card, and test that GPU acceleration is available.
Why is the speed on quadro gv100 slower than gtx 1660ti? It takes 240 seconds to train an epoch when conducting experiments on Quadro gv100.
It takes 37 seconds to train an epoch in the experiment on gtx 1660. The performance of Quqdro gv100 should be better than that of gtx 1660ti.
Why did I get a completely different point of view when I went to the lab? quadro gv100 is so much slower than gtx 1660ti.
What conditions should be met to turn on gpu acceleration on quadro gv100? But I have verified that gpu acceleration is available

The operating system of Quadro GV100 is windows server 2016, and the operating system of gtx 1660ti is windows10

**Comment by ravikyram** [comment]
@lihualilee 

Which TF version you are using?
Thanks!

**Comment by lihualilee** [comment]
@ Quadro GV100:TF2.2
gtx 1660:TF2.0

**Comment by sanjoy** [comment]
@lihualilee Can you try re-running with the same TF version on GV100 and GTX?  Right now it isn't clear if this is a regression from 2.0 to 2.2 or something else.  Also, can you try using tf-nightly?  That will tell us if the problem is fixed at HEAD.

Finally, have you looked into the issue with the [TensorFlow profiler](https://www.tensorflow.org/tensorboard/tensorboard_profiling_keras)?

**Comment by lihualilee** [comment]
@sanjoy 我不认为这个是tf版本2.2和2.0之间的问题。但是我在训练代码时时，显卡的性能一直不变，和空闲状态一样，一直是P2等级，并且功率也是保持不变的，30w/250w，显卡的温度也并没有上升。事实上，如果gpu在工作的话，那么它的温度会迅速上升，这一点很奇怪
127/5000
I don't think this is a problem between tf versions 2.2 and 2.0. However, when I was training the code, the performance of the graphics card remained unchanged. Like the idle state, it was always P2 level, and the power remained unchanged. At 30w/250w, the temperature of the graphics card did not rise. In fact, if the gpu is working, its temperature will rise rapidly, which is very strange

**Comment by lihualilee** [comment]
@sanjoy 
I don't think this is a problem between tf versions 2.2 and 2.0. However, when I was training the code, the performance of the graphics card remained unchanged. Like the idle state, it was always P2 level, and the power remained unchanged. At 30w/250w, the temperature of the graphics card did not rise. In fact, if the gpu is working, its temperature will rise rapidly, which is very strange.

**Comment by achintya-7** [comment]
Wait I wanted to ask
Does 1660ti even have cuda cores for tensorflow
